src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+			case ColorSwitchCommand.SupportedGet:
+			case ColorSwitchCommand.Get:
+			case ColorSwitchCommand.Set:
+				return true; // This is mandatory
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	public async supportedGet() {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.SupportedGet,
+		);
+
+		const cc = new ColorSwitchCCSupportedGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = (await this.driver.sendCommand<
+			ColorSwitchCCSupportedReport
+		>(cc))!;
+		return {
+			supportsWarmWhite: response.supportsWarmWhite,
+			supportsColdWhite: response.supportsColdWhite,
+			supportsRed: response.supportsRed,
+			supportsGreen: response.supportsGreen,
+			supportsBlue: response.supportsBlue,
+			supportsAmber: response.supportsAmber,
+			supportsCyan: response.supportsCyan,
+			supportsPurple: response.supportsPurple,
+			supportsIndex: response.supportsIndex,
+		};
+	}
+
+	public async get(colorComponent: ColorComponent) {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Get);
+
+		const cc = new ColorSwitchCCGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent: colorComponent,
+		});
+		const response = (await this.driver.sendCommand<ColorSwitchCCReport>(
+			cc,
+		))!;
+		return {
+			colorComponent: response.colorComponent,
+			value: response.value,
+		};
+	}
+
+	public async set(colorTable: ColorTable) {
+		this.assertSupportsCommand(ColorSwitchCommand, ColorSwitchCommand.Set);
+
+		const cc = new ColorSwitchCCSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			...colorTable,
+		});
+
+		// TODO: No response from this?
+		await this.driver.sendCommand(cc);
+	}
+}
+
+@commandClass(CommandClasses["Color Switch"])
+@implementedVersion(2)
+export class ColorSwitchCC extends CommandClass {
+	declare ccCommand: ColorSwitchCommand;
+
+	// TODO: Interview?
+}
+
+@CCCommand(ColorSwitchCommand.SupportedReport)
+export class ColorSwitchCCSupportedReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+
+		// Don't know what order these come in.
+		//  Blindly assuming its in order of payload,
+		//  but the table seems to be big-endian.
+		this._supportsWarmWhite = Boolean(this.payload[0] & 1);
+		this._supportsColdWhite = Boolean(this.payload[0] & 2);
+		this._supportsRed = Boolean(this.payload[0] & 4);
+		this._supportsGreen = Boolean(this.payload[0] & 8);
+		this._supportsBlue = Boolean(this.payload[0] & 16);
+		this._supportsAmber = Boolean(this.payload[0] & 32);
+		this._supportsCyan = Boolean(this.payload[0] & 64);
+		this._supportsPurple = Boolean(this.payload[0] & 128);
+		this._supportsIndex = Boolean(this.payload[2] & 1);
+
```

Comment:
You're on the right track, bit masks are Big-Endian.
Variable length usually means that future specs might add additional bytes to the end. The check for at least 2 bytes should suffice, since the command will always contain at least 2.

I have a helper method for parsing and serializing variable-length bit masks. Assuming there won't be any different fields added after it and there won't be any gaps in the bit mask, you can do this:

```ts
const bitMask = parseBitMask(this.payload).map(Boolean);
[
  this._supportsWarmWhite,
  this._supportsColdWhite,
  this._supportsRed,
  this._supportsGreen,
  this._supportsBlue,
  this._supportsAmber,
  this._supportsCyan,
  this._supportsPurple,
  this._supportsIndex
] = bitMask;
```